Qrisp on Python 3.13 leads to multiple unit test failures

I came across this while working on #664 

I added Python 3.13 to the testing pipeline in [30344d9](https://github.com/eclipse-qrisp/Qrisp/pull/664/commits/30344d9cbde9ce5fb5441708dce45330287cceb6) where there were multiple unit test failures. If I compare the installed packages for version 3.11 and 3.13, I do not see any differences. This make me think there is some change in how Python 3.13 handles iteration.

```
FAILED tests/algorithms_tests/qaoa_tests/test_QAOAMkCS.py::test_mkcs_G1e2c - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/qaoa_tests/test_QAOAMkCS.py::test_mkcs_5nodes - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/qaoa_tests/test_QAOAQUBO.py::test_QUBO - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/qaoa_tests/test_QAOAmaxCut.py::test_maxcut - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/test_backtracking.py::test_backtracking - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/test_grovers_algorithm.py::test_grovers_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/algorithms_tests/test_grovers_algorithm.py::test_grovers_list_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_array_plus_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_array_plus_scalar - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_array_plus_numpy - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_array_minus_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_array_multiply_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_in_place_addition - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_arithmetic_validation.py::TestArithmeticExecution::test_in_place_with_numpy_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-quantum-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; array RHS-classical-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-quantum-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumFloat; scalar RHS-classical-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-quantum-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[Signed QuantumFloat; array RHS-classical-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-quantum-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; array RHS-classical-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-quantum-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-add] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-sub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-mul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-eq] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-ne] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-gt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-ge] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-lt] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_ops[QuantumModulus; scalar RHS-classical-le] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_bool_ops[and_] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_bool_ops[or_] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_bool_ops[xor] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[Signed QuantumFloat; array RHS-quantum-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[Signed QuantumFloat; array RHS-quantum-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[Signed QuantumFloat; array RHS-classical-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[Signed QuantumFloat; array RHS-classical-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[Signed QuantumFloat; array RHS-classical-imul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumFloat; scalar RHS-quantum-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumFloat; scalar RHS-quantum-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumFloat; scalar RHS-classical-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumFloat; scalar RHS-classical-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumFloat; scalar RHS-classical-imul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; array RHS-quantum-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; array RHS-quantum-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; array RHS-classical-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; array RHS-classical-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; array RHS-classical-imul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; scalar RHS-quantum-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; scalar RHS-quantum-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; scalar RHS-classical-iadd] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; scalar RHS-classical-isub] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_wise_inplace_ops[QuantumModulus; scalar RHS-classical-imul] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_all[All True, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_all[One False, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_all[All True, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_all[One False, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_any[All True, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_any[One False, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_any[All True, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_any[One False, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_element_eq - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_inject_add - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_modular_multiplication - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_get_measurement_returns_mapping - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_get_measurement_compatible_with_batched_backend - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/quantum_array/test_quantum_array.py::test_quantum_array_injection - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/test_conditional_environments_example.py::test_conditional_environments_example - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/test_hello_world.py::test_hello_world - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/core_tests/test_string_test.py::test_string_test - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/jax_tests/test_jasp_QAE.py::test_jasp_QAE_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_amplitude_amplification.py::test_jasp_amplitude_amplification_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_amplitude_amplification.py::test_jasp_amplitude_amplification_oblivious - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_grovers_algorithm.py::test_jasp_grovers_algorithm - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_all[All True, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_all[One False, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_all[All True, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_all[One False, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_any[All True, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_any[One False, axis 0] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_any[All True, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_jasp_quantum_array.py::test_quantum_array_any[One False, axis 1] - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_qubit_array_fusion.py::test_qubit_array_fusion - TypeError: 'QuantumArrayIterator' object is not iterable
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
FAILED tests/jax_tests/test_shift_functions.py::test_cyclic_shift_quantum_array_static - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/jax_tests/test_shift_functions.py::test_cyclic_shift_quantum_array_shift_2_static - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/arithmetic_tests/test_inpl_matrix_multiplication_example.py::test_inpl_matrix_multiplication_example - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/arithmetic_tests/test_matrix_multiplication_example.py::test_matrix_multiplication_example - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/arithmetic_tests/test_quantum_arithmetic.py::test_quantum_arithmetic - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_QAE.py::test_QAE_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_amplitude_amplification.py::test_amplitude_amplification_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_amplitude_amplification.py::test_amplitude_amplification_oblivious - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_cycling_function.py::test_cycling_function - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_reflection.py::test_reflection_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
FAILED tests/primitives_tests/test_reflection.py::test_reflection_list_quantum_varaible_quantum_array - TypeError: 'QuantumArrayIterator' object is not iterable
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ authors = [
 ]
 description = "A high-level quantum programming language"
 readme = "README.md"
-requires-python = ">=3.11, <3.13"
+requires-python = ">=3.11"
 classifiers = [
     "Intended Audience :: Science/Research",
     "License :: OSI Approved :: Eclipse Public License 2.0 (EPL-2.0)",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ authors = [
 ]
 description = "A high-level quantum programming language"
 readme = "README.md"
-requires-python = ">=3.11"
+requires-python = ">=3.11, <3.14"
 classifiers = [
     "Intended Audience :: Science/Research",
     "License :: OSI Approved :: Eclipse Public License 2.0 (EPL-2.0)",

--- a/setup.py
+++ b/setup.py
@@ -34,5 +34,5 @@ setuptools.setup(
     ],
     package_dir={"": "src"},
     packages=setuptools.find_packages(where="src"),
-    python_requires=">=3.11",
+    python_requires=">=3.11, <3.14",
 )

--- a/setup.py
+++ b/setup.py
@@ -34,5 +34,5 @@ setuptools.setup(
     ],
     package_dir={"": "src"},
     packages=setuptools.find_packages(where="src"),
-    python_requires=">=3.11, <3.13",
+    python_requires=">=3.11",
 )

--- a/src/qrisp/core/quantum_array.py
+++ b/src/qrisp/core/quantum_array.py
@@ -2134,6 +2134,9 @@ class QuantumArrayIterator:
         self.qa = qa
         self.counter = -1
 
+    def __iter__(self):
+        return self
+
     def __next__(self):
         self.counter += 1
         if self.counter >= self.qa.size:

